sklearn/feature_selection/_base.py

```diff
@@ -98,7 +98,7 @@ def _transform(self, X):
                 " too noisy or the selection test too strict.",
                 UserWarning,
             )
-            return np.empty(0).reshape((X.shape[0], 0))
+            return np.empty(0, dtype=X.dtype).reshape((X.shape[0], 0))
```

Comment:
We need to update or add test to check this. I do not think the common test will result in no features selected.

Reply:
Added a test under `test_invalid_k` in `test_feature_select.py` since using `k_best` w/ `param=0` was the most straightforward way of ensuring an empty support.

Reply:
Added a test in `test_feature_select.py`